### PR TITLE
New package: gamemode-1.6.1

### PR DESCRIPTION
**Pull request by kedodrill**
Closes #19288, #22291, #19281, #13246

#### New package
- This new package conforms to the [quality requirements](https://github.com/void-linux/void-packages/blob/master/Manual.md#quality-requirements): Yes

#### Local build testing
- I built this PR locally for my native architecture, x86_64
- I built this PR locally for these architectures (if supported. mark crossbuilds):
  - i686
  - x86_64-musl

### CODE REVIEW

**Comment by RicArch97** [comment]
Consider bumping to version 1.6.1, which is the lastest currently

File: common/shlibs [hunk 1]
```diff
@@ -3597,6 +3597,8 @@ libgnuradio-nrsc5-1.0.0git.so.0.0.0 gnuradio-nrsc5-1.0.0_1
 libvkd3d-utils.so.1 vkd3d-1.1_1
 libvkd3d-shader.so.1 vkd3d-1.3_1
 libvkd3d.so.1 vkd3d-1.1_1
+libgamemode.so.0 libgamemode-1.6_1
+libgamemodeauto.so.0 libgamemode-1.6_1
```
Comment (RicArch97) [inline common/shlibs:3601]:
```suggestion
libgamemode.so.0 libgamemode-1.6.1_1
libgamemodeauto.so.0 libgamemode-1.6.1_1
```
Perhaps these need to be bumped as well

Comment (kedodrill) [reply common/shlibs:3601]: oh yep, totally forgot about those

**Comment by frankfutlg** [comment]
how do i add gamemode to void packages?

**Comment by RicArch97** [comment]
> how do i add gamemode to void packages?

You can do the following to test this PR locally;
In your void-packages folder, run:
```console
$ git fetch origin pull/36564/head:gamemode
$ git checkout gamemode
```
And then `./xbps-src pkg gamemode` to build it

**Comment by frankfutlg** [comment]
thanks!! it built and installed succesfully (i used xi after building), but, just to confirm, to build it in x86 (in order to use gamemode on lutris and steam i need the x86 version too), i make a new masterdir in x86 with ./xbps-src -m masterdir-x86 binary-bootstrap i686 and then i build it with ./xbps-src -m masterdir-x86 pkg gamemode, right?

**Comment by kedodrill** [comment]
@frankfutlg Are you essentially trying to install what would be the `gamemode-32bit` package? I'm actually not sure on that, because using another masterdir to compile for x86 would result in a 32bit package with a 32bit binary. I believe void packages *-32bit packages as just library files and they don't include binaries, so I don't think that would work. I'm unsure how you would locally build the 32bit version...some clarity on that would be good if anyone can provide.

**Comment by RicArch97** [comment]
> i make a new masterdir in x86 with ./xbps-src -m masterdir-x86 binary-bootstrap i686 and then i build it with ./xbps-src -m masterdir-x86 pkg gamemode, right?

This is correct. You'd basically install `gamemode`, `libgamemode` and optionally `libgamemode-devel` for 64 bit, and for 32 bit you can install `libgamemode-32bit` and optionally `libgamemode-devel-32bit` after building.

**Comment by frankfutlg** [comment]
well, i have a problem, after installing it i have not tested it to see if it works, and i tried doing gamemoded-t and if i am remembering correctly it gave a reaper thread error, and after it finished, a process called dbus-launch spammed non-stop, and it ended up overloading my ram, i had to force-shutdown and then boot again.

i also tried to play steam games with it, and when i launched it, i waited some seconds and it closed.

is there a fix for those problems?

**Comment by RicArch97** [comment]
> well, i have a problem, after installing it i have not tested it to see if it works, and i tried doing gamemoded-t and if i am remembering correctly it gave a reaper thread error, and after it finished, a process called dbus-launch spammed non-stop, and it ended up overloading my ram, i had to force-shutdown and then boot again.
> 
> i also tried to play steam games with it, and when i launched it, i waited some seconds and it closed.
> 
> is there a fix for those problems? and sorry for the bad english, its not my first language.

I cannot reproduce this (using default config), so likely not an issue with the package. Also works fine for me on Steam and Lutris.

```console
$ gamemoded -t
: Loading config
Loading config file [/usr/share/gamemode/gamemode.ini]
: Running tests

:: Basic client tests
:: Passed

:: Dual client tests
gamemode request succeeded and is active
Quitting by request...
:: Passed

:: Gamemoderun and reaper thread tests
...Waiting for child to quit...
...Waiting for reaper thread (reaper_frequency set to 5 seconds)...
:: Passed

:: Supervisor tests
:: Passed

:: Feature tests
::: Verifying CPU governor setting
::: Passed
::: Verifying Scripts
::: Passed (no scripts configured to run)
::: Verifying GPU Optimisations
::: Passed (gpu optimisations not configured to run)
::: Verifying renice
::: Passed (no renice configured)
::: Verifying ioprio
::: Passed
:: Passed

: All Tests Passed!
```

**Comment by frankfutlg** [comment]
tbh, i really dont know why i have this issue, one time i used gentoo, and gamemode was available to install with elogind, and i had the same issue

**Comment by MCPEngu** [comment]
Well even on default config, I still have dbus-launch process spammed when launching gamemode.

**Comment by MCPEngu** [comment]
nevermind, adding `dbus-launch --exit-with-session` on `/usr/share/xsessions` desktop exec file worked for me

**Comment by frankfutlg** [comment]
> nevermind, adding `dbus-launch --exit-with-session` in `/usr/share/xsessions` desktop exec file worked for me

well, i tried doing that (adding it on my .xinitrc since i dont use display managers), and it didn't work.
it still spams dbus-launch after gamemoded -t finishes and gamemode still doesn't work

**Comment by frankfutlg** [comment]
no! nevermind.. what i needed to do was just executing gamemode with that parameter. now it works!

 $ dbus-launch --exit-with-session gamemoded -t                                                                                                                                  
: Loading config
Loading config file [/usr/share/gamemode/gamemode.ini]
: Running tests

:: Basic client tests
:: Passed

:: Dual client tests
gamemode request succeeded and is active
Quitting by request...
:: Passed

:: Gamemoderun and reaper thread tests
...Waiting for child to quit...
...Waiting for reaper thread (reaper_frequency set to 5 seconds)...
:: Passed

:: Supervisor tests
:: Passed

:: Feature tests
::: Verifying CPU governor setting
::: Passed
::: Verifying Scripts
::: Passed (no scripts configured to run)
::: Verifying GPU Optimisations
::: Passed (gpu optimisations not configured to run)
::: Verifying renice
::: Passed (no renice configured)
::: Verifying ioprio
::: Passed
:: Passed

: All Tests Passed

@MCPEngu thanks mate!!!

**Comment by MCPEngu** [comment]
> no! nevermind.. what i needed to do was just executing gamemode with that parameter. now it works!
> 
> $ dbus-launch --exit-with-session gamemoded -t : Loading config Loading config file [/usr/share/gamemode/gamemode.ini] : Running tests
> 
> :: Basic client tests :: Passed
> 
> :: Dual client tests gamemode request succeeded and is active Quitting by request... :: Passed
> 
> :: Gamemoderun and reaper thread tests ...Waiting for child to quit... ...Waiting for reaper thread (reaper_frequency set to 5 seconds)... :: Passed
> 
> :: Supervisor tests :: Passed
> 
> :: Feature tests ::: Verifying CPU governor setting ::: Passed ::: Verifying Scripts ::: Passed (no scripts configured to run) ::: Verifying GPU Optimisations ::: Passed (gpu optimisations not configured to run) ::: Verifying renice ::: Passed (no renice configured) ::: Verifying ioprio ::: Passed :: Passed
> 
> : All Tests Passed
> 
> @MCPEngu thanks mate!!!

No problem :D

**Pull request merged**